### ISSUE TITLE
Unable to find current revision in submodule path 'src/lib/ecl'

### ISSUE THREAD

**Issue by potato77** [PX4/Firmware#6697]
Hello,

I used the cmd git submodule update --init --recursive to update all submodules of px4 .

and the error is:

fatal: Needed a single revision
Unable to find current revision in submodule path 'src/lib/ecl'

How to fix it? Thank you in advance!

any help would be appreciated!

**Comment by mhkabir** [comment]
Please run `make submodulesclean`

**Comment by potato77** [comment]
$ make submodulesclean

Usage: git submodule [--quiet] add [-b branch] [-f|--force] [--reference <repository>] [--] <repository> [<path>]          or: git submodule [--quiet] status [--cached] [--recursive] [--] [<path>...]                                            or: git submodule [--quiet] init [--] [<path>...]                                                                       or: git submodule [--quiet] update [--init] [-N|--no-fetch] [-f|--force] [--rebase] [--reference <repository>] [--merge] [--recursive] [--] [<path>...]                                                                                         or: git submodule [--quiet] summary [--cached|--files] [--summary-limit <n>] [commit] [--] [<path>...]                  or: git submodule [--quiet] foreach [--recursive] <command>                                                             or: git submodule [--quiet] sync [--] [<path>...]                                                                    make: *** [submodulesclean] Error 1

**Comment by mhkabir** [comment]
That's pretty odd. What operating system and version are you using? 

@dagar any suggestions?

**Comment by potato77** [comment]
windows 10

**Comment by dagar** [comment]
@potato77 have you successfully built PX4 on windows before? What branch are you on?

**Comment by potato77** [comment]
@dagar no. This is my first time to install the development environment.

**Comment by dagar** [comment]
Can you walk us through what you've done so far, which tools you've installed. It's not clear to me if you have a really old version of git or there's some Windows path or line ending issue.

If you're able to use Linux (Ubuntu) or OS X for development you'll have a much easier time. It's possible to get this working in Windows, but much slower and poorly supported.

**Comment by potato77** [comment]
@dagar thank u very much
I am trying to install the development environment on the VM unbuntu.

**Comment by mhkabir** [comment]
I believe we could eventually create a setup that works on Bash for Windows 10, thus avoiding all the issues we're currently having. 

I had a quick look-see and it seems like it'll be easy to do.

**Comment by dagar** [comment]
That would certainly be nice. If there's anything to do to get it working we can then keep it working by adding a build on https://www.appveyor.com/.

**Comment by potato77** [comment]
@mhkabir any instruction to Bash for WIN10? How to build PX4?

**Comment by mhkabir** [comment]
@potato No instructions yet. I don't have a Windows 10 machine running at the moment. I can however assist you in bringing it up remotely, later today. Skype me : 'kabiriris'.

**Comment by mhkabir** [comment]
Start by installing Bash if you don't have it already : https://www.howtogeek.com/249966/how-to-install-and-use-the-linux-bash-shell-on-windows-10/

**Comment by mhkabir** [comment]
@dagar I don't think Appveyor has support for Bash on Ubuntu yet : https://github.com/appveyor/ci/issues/1295

**Comment by mhkabir** [comment]
If anyone running Windows would like to help me with testing, it would be awesome. You can email me or ping me on Skype, and I'll guide you through the process. 
We can then update Devguide with a working set of instructions for everyone to use.

**Comment by dagar** [comment]
This works for me.
